File: Python/lecturer.py
from civitasacademic import CivitasAcademic # Import class CivitasAcademic dari file CivitasAcademic.py
from major import Major # Import class Major dari file major.py

class Lecturer(CivitasAcademic): # Deklarasi class Lecturer (child class dari CivitasAcademic)
    __offRegNumber = 0 # Untuk menyimpan nomor induk dosen, atribut berjenis private
    __teritaryEducation = "" # Untuk menyimpan pendidikan terakhir dosen, atribut berjenis private
    __proficiency = "" # Untuk menyimpan keahlian dosen, atribut berjenis private
    __majorObject = Major("", "", []) # Untuk menyimpan jurusan dosen, atribut berjenis private

    def __init__(self, identity, name, gender, university, email, faculty, offRegNumber, teritaryEducation, proficiency, major): # Membuat constructor
        super().__init__(identity, name, gender, university, email, faculty) # Menginisialisasi setiap atribut yang ada di dalam konstruktor parent class CivitasAcademic
        self.__offRegNumber = offRegNumber # Menginisialisasi atribut __offRegNumber
        self.__teritaryEducation = teritaryEducation # Menginisialisasi atribut __teritaryEducation
        self.__proficiency = proficiency # Menginisialisasi atribut __proficiency
        self.__majorObject = major # Menginisialisasi atribut __majorObject

    def setOffRegNumber(self, offRegNumber): # Setter untuk atribut __offRegNumber
        self.__offRegNumber = offRegNumber # Menginisialisasi atribut __offRegNumber 

    def setTeritaryEducation(self, teritaryEducation): # Setter untuk atribut __teritaryEducation
        self.__teritaryEducation = teritaryEducation # Menginisialisasi atribut __teritaryEducation

    def setProficiency(self, proficiency): # Setter untuk atribut __proficiency
        self.__proficiency = proficiency # Menginisialisasi atribut __proficiency

    def getOffRegNumber(self): # Getter untuk atribut __offRegNumber
        return self.__offRegNumber # Mengembalikan nilai atribut __offRegNumber
    
    def getTeritaryEducation(self): # Getter untuk atribut __teritaryEducation
        return self.__teritaryEducation # Mengembalikan nilai atribut __teritaryEducation
    
    def getProficiency(self): # Getter untuk atribut __proficiency
        return self.__proficiency # Mengembalikan nilai atribut __proficiency
    
    def getMajor(self): # Getter untuk atribut __majorObject
        return self.__majorObject # Mengembalikan nilai atribut __majorObject

File: C++/Lecturer.cpp
// Import library yang dibutuhkan
#include <bits/stdc++.h>

using namespace std; // Menggunakan standard namespace

class Lecturer : public CivitasAcademic // Membuat class Lecturer (child) yang mengextends ke class CivitasAcademic (parent)
{
private:                      // Membuat atribut berjenis private
    int offRegNumber;         // Untuk menyimpan NIP dosen
    string teritaryEducation; // Untuk menyimpan pendidikan terakhir dosen
    string proficiency;       // Untuk menyimpan keahlian dosen
    Major majorObject;        // Untuk menyimpan objek dari class Major

public:
    /* Konstruktor */
    Lecturer() : CivitasAcademic() // Membuat konstruktor tanpa isian, class yang di-extends akan diinstansiasi terlebih dahulu
    {
        offRegNumber = 0;
        teritaryEducation = "";
        proficiency = "";
        majorObject = Major();
    }

    Lecturer(long long identity, string name, char gender, string university, string faculty, string email, int offRegNumber, string teritaryEducation, string proficiency, Major major) : CivitasAcademic(identity, name, gender, university, faculty, email) // Membuat konstruktor dengan isian dari parameter
    {
        this->offRegNumber = offRegNumber;
        this->teritaryEducation = teritaryEducation;
        this->proficiency = proficiency;
        this->majorObject = major;
    }

    /* Setter dan Getter untuk setiap atribut di dalam class Lecturer */
    void setOffRegNumber(int offRegNumber)
    {
        this->offRegNumber = offRegNumber;
    }

    void setTeritaryEducation(string teritaryEducation)
    {
        this->teritaryEducation = teritaryEducation;
    }

    void setProficiency(string proficiency)
    {
        this->proficiency = proficiency;
    }

    int getOffRegNumber()
    {
        return this->offRegNumber;
    }

    string getTeritaryEducation()
    {
        return this->teritaryEducation;
    }

    string getProficiency()
    {
        return this->proficiency;
    }

    Major getMajorObject()
    {
        return this->majorObject;
    }

    /* Destruktor */
    ~Lecturer() // Membuat destruktor untuk menghapus semua objek yang telah dibuat
    {
    }
};